admin-auth/src/main/kotlin/com/epam/drill/admin/auth/config/SecurityConfig.kt

```diff
@@ -118,39 +153,43 @@ class SecurityConfig(override val di: DI) : DIAware {
 }
 
 fun Routing.oauthRoutes() {
+    val oauthConfig by di().instance<OAuthConfig>()
+
     authenticate("oauth") {
         get("/oauth/login") {
-            // Redirects to "authorizeUrl" if authentication fails
-            // or redirect to root otherwise
-            call.respondRedirect("/")
-        }
-        get("/oauth/logout") {
-            call.principal<OAuthAccessTokenResponse.OAuth2>()?.let { principal ->
-                val httpClient = HttpClient(Apache)
-                httpClient.post<HttpResponse>("http://localhost:8080/realms/master/protocol/openid-connect/logout") {
-                    headers {
-                        append(HttpHeaders.Authorization, "Bearer ${principal.accessToken}")
-                    }
-                }
-            }
+            // Redirects to "authorizeUrl" automatically
         }
         get("/oauth/callback") {
-            val principal: OAuthAccessTokenResponse.OAuth2? = call.principal()
-            if (principal == null) {
+            val oauthPrincipal: OAuthAccessTokenResponse.OAuth2? = call.principal()
+            if (oauthPrincipal == null) {
                 call.unauthorizedError()
                 return@get
             }
 
-            call.response.cookies.append(Cookie("jwt", principal.accessToken, httpOnly = true, path = "/"))
-            call.respondRedirect("/") //TODO configurable redirect target
+            val user = getUserInfo(oauthConfig.userInfoUrl, oauthPrincipal.accessToken)
+            if (user == null) {
+                call.unauthorizedError()
```

Comment:
can't we just do `throw UnauthorizedError()` and use centralized error handling / status pages to handle that?

Reply:
Oauth2 routes are attached outside of the main routes and common error handlers for them may not work